dockerfiles/mlrun-api/Dockerfile

```diff
@@ -41,21 +44,29 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
   ca-certificates \
  && rm -rf /var/lib/apt/lists/*
 
-# Install uv
-ARG MLRUN_UV_VERSION=0.5.5
-ADD https://astral.sh/uv/${MLRUN_UV_VERSION}/install.sh /uv-installer.sh
+ENV UV_SYSTEM_PYTHON=true UV_LINK_MODE=copy
 
-RUN sh /uv-installer.sh && rm /uv-installer.sh
+WORKDIR /mlrun
 
-ENV PATH="/root/.local/bin/:$PATH"
-ENV UV_SYSTEM_PYTHON=true
+RUN --mount=from=uv-image,source=/uv,target=/bin/uv \
+    --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=dockerfiles/mlrun-api/locked-requirements.txt,target=locked-requirements.txt \
+    uv pip sync --require-hashes locked-requirements.txt
```

Comment:
if this ( + the UV image initialization) is going to be the same in all mlrun images, perhaps we can create a custom base image that will be used as a `FROM` for all of them?
not critical for this PR.

Reply:
that needs to have a bigger refactor but indeed could be optimized using bytecode compilation